### PR TITLE
nydus-image: fix rafs v6 to get file's parent inode

### PR DESCRIPTION
**Pull request by sctb512**
Currently, the parent of file is None,
we can set the parent inode by traversing root directory.
This PR is addressing #768 

Signed-off-by: Bin Tang <tangbin.bin@bytedance.com>

### CODE REVIEW

**Comment by jiangliu** [comment]
I have different thought here. It's a little heavy to enhance get_inode_ext() to support such a case, it may  potentially cause serious performance issue. So I would prefer only to change `nydus-image icheck` implementation.

File: rafs/src/metadata/direct_v6.rs [hunk 1]
```diff
@@ -356,6 +356,29 @@ impl RafsSuperInodes for DirectSuperBlockV6 {
...
+                        return Ok(Arc::new(inode));
+                    }
+
+                    if child.is_dir() {
+                        let child_inode = self.get_inode(child.ino(), false)?;
+                        stack.push_back(child_inode);
+                    }
+                }
+            }
+        }
```
Comment (changweige) [inline rafs/src/metadata/direct_v6.rs:381]: We'd better make this optional in order not to affect nydusd runtime

File: rafs/src/metadata/direct_v6.rs [hunk 2]
```diff
@@ -356,6 +356,29 @@ impl RafsSuperInodes for DirectSuperBlockV6 {
...
         }
+
+        let root_ino = self.root_ino();
+        let root = self.get_inode(root_ino, false)?;
+        let mut stack = VecDeque::new();
+        if root.is_dir() {
+            stack.push_back(root);
+        }
+        while !stack.is_empty() {
+            let parent = stack.pop_front().unwrap();
```
Comment (changweige) [inline rafs/src/metadata/direct_v6.rs:367]: Don't directly unwrap unless is safe, which should be commented

File: rafs/src/metadata/direct_v6.rs (hunk 1 above)
Comment (sctb512) [reply rafs/src/metadata/direct_v6.rs]: Fixed.

**Pull request closed without merging**